std/process.d

```diff
@@ -2046,6 +2113,31 @@ enum Config
     stderrPassThrough = 128,
 }
 
+/**
+A sturct that controls the behavior of process creation functions in this module.
+*/
+struct AdvancedConfig
+{
+    /**
+    A function type that notifies error in forked process.
+    */
+    alias AbortOnError = void delegate() nothrow @nogc @safe;
```

Comment:
We are forcing the user to provide a pre-exec function that doesn't throw, does not allocate and is safe. This might be too restrictive.

Reply:
The alternative would be to simply declare the delegate without the attributes. That way the user can provide both @safe and non-@safe delegates.

Reply:
I agree with it is too restrictive.

I will remove attributes except safe.
Accepting unsafe delegates breaks memory safety of `spawnProcess` because it is currently marked as trusted and thus the unsafe delegates are accidentally treated as safe.

Reply:
> We are forcing the user to provide a pre-exec function that doesn't throw, does not allocate and is safe. This might be too restrictive.

I just realized that I was talking about the restrictions that would be implicit with using `vfork` or something similar, which we do not actually use right now.

I see that an attempt has already been made to use `vfork`, which did not go through: https://github.com/dlang/phobos/pull/3476

However, I'm not sure if we should completely resign ourselves to using `fork` for `std.process` either.